core/src/main/scala/org/http4s/Entity.scala

```diff
@@ -16,19 +16,58 @@
 
 package org.http4s
 
-import cats.Monoid
+import fs2.{Stream, Chunk}
+import cats._
 import cats.syntax.all._
 
-final case class Entity[+F[_]](body: EntityBody[F], length: Option[Long] = None)
+sealed trait Entity[F[_]]{
+  def body: Stream[F, Byte]
+  def length: Option[Long]
+  def translate[G[_]](fk: F ~> G): Entity[G]
+  def covary[G[x] >: F[x]]: Entity[G] = translate[G](cats.arrow.FunctionK.id[F].widen[G]) 
+}
 
 object Entity {
+
+  case class Strict[F[_]](chunk: Chunk[Byte]) extends Entity[F]{
+    def body: Stream[F, Byte] = Stream.chunk(chunk)
+    def length: Option[Long] = chunk.size.toLong.some
+    def translate[G[_]](fk: F ~> G): Entity[G] = Strict(chunk)
```

Comment:
Can't we just cast this instead of allocating a new one?

Reply:
Yes. I just hate writing casts on a first draft where things aren't yet compiling right.